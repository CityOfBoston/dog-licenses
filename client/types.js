// @flow

import type { SearchDogLicensesQuery } from './queries/graphql-types';
import type { DogLicenseCreateMutation } from './queries/graphql-types';

export type DogLicenseSearchResults = $PropertyType<
  $PropertyType<SearchDogLicensesQuery, 'dogLicenses'>,
  'search',
>;

export type DogLicenseCreateResults = $PropertyType<
  $PropertyType<DogLicenseCreateMutation, 'dogLicenseCreate'>,
  'firstName',
>;

export type DogLicense = $ArrayElement<DogLicenseSearchResults>;
