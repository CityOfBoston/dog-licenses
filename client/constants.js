// @flow

export const COLORS = {
  Agouti: 'AGOUTI',
  Apricot: 'APRICOT',
  'Black Brindle': 'BL BRINDLE',
  Black: 'BLACK',
  'Black Smoke': 'BLK SMOKE',
  Blue: 'BLUE',
  'Blue Merle': 'BLUE MERLE',
  'Blue PT': 'BLUE PT',
  'Blue Tick': 'BLUE TICK',
  'Blue Tiger': 'BLUE TIGER',
  'Brown Brindle': 'BR BRINDLE',
  'Brown Merle': 'BRN MERLE',
  Brown: 'BROWN',
  Buff: 'BUFF',
  Chocolate: 'CHOCOLATE',
  Cream: 'CREAM',
  Fawn: 'FAWN',
  Gold: 'GOLD',
  Gray: 'GRAY',
  Green: 'GREEN',
  'Lilac PT': 'LILAC PT',
  Liver: 'LIVER',
  'Liver Tick': 'LIVER TICK',
  Orange: 'ORANGE',
  'Orange Tiger': 'ORANGE TIGER',
  Pink: 'PINK',
  Purple: 'PURPLE',
  Red: 'RED',
  'Red Merle': 'RED MERLE',
  'Red Tick': 'RED TICK',
  Muddle: 'MUDDLE',
  Sable: 'SABLE',
  'Seal PT': 'SEAL PT',
  Silver: 'SILVER',
  Tan: 'TAN',
  Tortie: 'TORTIE',
  Tricolor: 'TRICOLOR',
  White: 'WHITE',
  'Yellow Brindle': 'Y BRINDLE',
  Yellow: 'YELLOW',
};

export const NEIGHBORHOODS = {
  Allston: 'ALLSTON',
  'Back Bay': 'BACK BAY',
  'BEACON HILL': 'Beacon Hill',
  Boston: 'BOSTON',
  Brighton: 'BRIGHTON',
  Charleston: 'CHARLESTON',
  Dorchester: 'DORCHESTER',
  'East Boston': 'EAST BOSTON',
  'Hyde Park': 'HYDE PARK',
  'Jamaica Plain': 'JAMAICA PLAIN',
  Mattapan: 'MATTAPAN',
  Readville: 'READVILLE',
  Roslindale: 'ROSLINDALE',
  Roxbury: 'ROXBURY',
  'South Boston': 'SOUTH BOSTON',
  'West Roxbury': 'WEST ROXBURY',
};

export const BREEDS = {
  'Boykin Spaniel': 'BOYKIN SPAN',
  'Cairn Terrier': 'CAIRN TERRIER',
  'Bull Terrier': 'BULL TERRIER',
  Beauceron: 'BEAUCERON',
  'Belgian Malnois': 'BELG MALINOIS',
  'Ibizan Hound': 'IBIZAN HOUND',
  'French Bulldog': 'FRENCH BULLDOG',
  'Queensland Heel': 'QUEENSLAND HEEL',
  'German Shephard Point': 'GERM SH POINT',
  'Karelian Bear': 'KARELIAN BEAR',
  'English Bulldog': 'ENG BULLDOG',
  Landseer: 'LANDSEER',
  Vizla: 'VIZSLA',
  'Cocker Spaniel': 'COCKER SPAN',
  'Schnauzer (Miniature)': 'SCHNAUZER MIN',
  'Mexican Hairless': 'MEX HAIRLESS',
  'Italian Greyhound': 'ITAL GREYHOUND',
  'Portuguese Water Dog': 'PORT WATER DOG',
  'Sealyham Terrier': 'SEALYHAM TERR',
  'St. Bernard (SMTH)': 'ST BERNARD SMTH',
  'Plott Hound': 'PLOTT HOUND',
  Otterhound: 'OTTERHOUND',
  'Shetland Sheepdog': 'SHETLD SHEEPDOG',
  'Fox Terrier (Wire-haired)': 'FOX TERR WIRE',
  Briard: 'BRIARD',
  'Norwich Terrier': 'NORWICH TERRIER',
  Greyhound: 'GREYHOUND',
  'Cavalier Spaniel': 'CAVALIER SPAN',
  'German Shephard': 'GERM SHEPHERD',
  'Spanish Water Dog': 'SPAN WATER DOG',
  'Wire-haired Pointer Griffon': 'WH PT GRIFFON',
  'Lhasa Apso': 'LHASA APSO',
  'Manchester Terrier': 'MANCHESTER TERR',
  'Rat Terrier': 'RAT TERRIER',
  'Chesapeake Bay Retriever': 'CHESA BAY RETR',
  'Schnauzer (Giant)': 'SCHNAUZER GIANT',
  'Bluetick Hound': 'BLUETICK HOUND',
  Akita: 'AKITA',
  Harrier: 'HARRIER',
  'Pharoah Hound': 'PHARAOH HOUND',
  'American Bulldog': 'AMER BULLDOG',
  'Shiba Inu': 'SHIBA INU',
  'Gordon Setter': 'GORDON SETTER',
  'Collie (Smooth)': 'COLLIE SMOOTH',
  'American Eskimo': 'AMER ESKIMO',
  'English Springer Spaniel': 'ENG SPRNGR SPAN',
  Bullmastiff: 'BULLMASTIFF',
  Lowchen: 'LOWCHEN',
  'Italian Spinone': 'SPINONE ITAL',
  'Great Dane': 'GREAT DANE',
  'Bruss Griffon': 'BRUSS GRIFFON',
  Maltese: 'MALTESE',
  'West Highland': 'WEST HIGHLAND',
  'Treeing Walker Coonhound': 'TR WALKER HOUND',
  Tosa: 'TOSA',
  'American Foxhound': 'AMER FOXHOUND',
  'Tibetan Spaniel': 'TIBETAN SPAN',
  Papillion: 'PAPILLON',
  Ragdoll: 'RAGDOLL',
  'Old Enlish Bulldog': 'OLD ENG BULLDOG',
  Leonberger: 'LEONBERGER',
  'Greater Swiss Mountain': 'GR SWISS MTN',
  'Dutch Sheepdog': 'DUTCH SHEEPDOG',
  'Bichon Frise': 'BICHON FRISE',
  'Tibetan Mastiff': 'TIBETAN MASTIFF',
  Havanese: 'HAVANESE',
  'Mini Pinscher': 'MIN PINSCHER',
  Himalayan: 'HIMALAYAN',
  'Poodle (Standard)': 'POODLE STND',
  'Fox Terrier (Smooth)': 'FOX TERR SMOOTH',
  'Dogue de Bordeaux': 'DOGUE DE BORDX',
  'Labrador Retriever': 'LABRADOR RETR',
  'Nova Scotia Duck Tolling Retriever': 'NS DUCK TOLLING',
  'Belgian Tervuren': 'BELG TERVUREN',
  'Welsh Terrier': 'WELSH TERRIER',
  'Japanese Chin': 'JAPANESE CHIN',
  Hovaward: 'HOVAWART',
  'Irish Water Spaniel': 'IRISH WATR SPAN',
  'Hound (Black/Tan)': 'BLACK/TAN HOUND',
  Brittany: 'BRITTANY',
  'Silky Terrier': 'SILKY TERRIER',
  Staffordshire: 'STAFFORDSHIRE',
  'Siberian Husky': 'SIBERIAN HUSKY',
  'Chihuahua (Long-haired)': 'CHIHUAHUA LH',
  'Bull Terrier (Mini)': 'BULL TERR MIN',
  Rottweiler: 'ROTTWEILER',
  'American Staffordshire Terrier': 'AMERICAN STAFF',
  'Black Russian Terrier': 'BL RUSSIAN TER',
  Kuvasz: 'KUVASZ',
  'Kerry Blue Terrier': 'KERRY BLUE TERR',
  Basenji: 'BASENJI',
  'Pit Bull': 'PIT BULL',
  Pomeranian: 'POMERANIAN',
  'Sussex Spaniel': 'SUSSEX SPAN',
  Bat: 'BAT',
  'Alaskan Klee Kai': 'ALASK KLEE KAI',
  'Norweigan Elkhound': 'NORW ELKHOUND',
  Beagle: 'BEAGLE',
  'English Cocker Spaniel': 'ENG COCKER SPAN',
  'Poodle (Toy)': 'POODLE TOY',
  'English Pointer': 'ENG POINTER',
  'English Foxhound': 'ENG FOXHOUND',
  Siamese: 'SIAMESE',
  'Basset Hound': 'BASSET HOUND',
  'Irish Terrier': 'IRISH TERRIER',
  'Scottish Terrier': 'SCOT TERRIER',
  'Dachshund (Long-haired)': 'DACHSHUND LH',
  'Dutch Shepherd': 'DUTCH SHEPHERD',
  Borzoi: 'BORZOI',
  'American Shepherd': 'AMER SH',
  'Alaskan Husky': 'ALASKAN HUSKY',
  'Alaskan Malamute': 'ALASK MALAMUTE',
  'Belgian Sheepdog': 'BELG SHEEPDOG',
  'Soft Coated Wheaten Terrier': 'SC WHEAT TERR',
  'Irish Setter': 'IRISH SETTER',
  'English Setter': 'ENG SETTER',
  'Glen of Imaal Terrier': 'GLEN OF IMAAL',
  Komondor: 'KOMONDOR',
  'Poodle (Mini)': 'POODLE MIN',
  'Bedlington Terrier': 'BEDLINGTON TERR',
  'Bouvier Flandres': 'BOUV FLANDRES',
  'German Wire-haired Pointer': 'GERM WH POINT',
  'Field Spaniel': 'FIELD SPANIEL',
  'Chinese Crested': 'CHINESE CRESTED',
  'American Pit Bull Terrier': 'AM PIT BULL TER',
  Pug: 'PUG',
  Affenpinscher: 'AFFENPINSCHER',
  'Welsh Springer Spaniel': 'WELSH SPR SPAN',
  'Black Mouth Cur': 'BLACK MOUTH CUR',
  'Shih Tzu': 'SHIH TZU',
  Bulldog: 'BULLDOG',
  'Schnauzer (Standard)': 'SCHNAUZER STAND',
  Akbash: 'AKBASH',
  Feist: 'FEIST',
  Pointer: 'POINTER',
  'Swedish Vallhund': 'SWED VALLHUND',
  'English Toy Spaniel': 'ENG TOY SPANIEL',
  'Dachshund (Wire-haired)': 'DACHSHUND WH',
  'Patterdale Terrier': 'PATTERDALE TERR',
  'Redbone Hound': 'REDBONE HOUND',
  Bloodhound: 'BLOODHOUND',
  Boerboel: 'BOERBOEL',
  'English Coonhound': 'ENG COONHOUND',
  'Welsh Corgi (Cardigan)': 'WELSH CORGI CAR',
  'Polish Lowland': 'POLISH LOWLAND',
  Jindo: 'JINDO',
  'Swiss Hound': 'SWISS HOUND',
  'Lakeland Terrier': 'LAKELAND TERR',
  'Rhodesian Ridgeback': 'RHOD RIDGEBACK',
  'Border Collie': 'BORDER COLLIE',
  'Coton de Tulear': 'COTON DE TULEAR',
  'Fox Terrier (Toy)': 'TOY FOX TERRIER',
  'Airedale Terrier': 'AIREDALE TERR',
  'Scottish Deerhound': 'SCOT DEERHOUND',
  Saluki: 'SALUKI',
  'Flat Coat Retriever': 'FLAT COAT RETR',
  Samoyed: 'SAMOYED',
  Boxer: 'BOXER',
  'Carolina Dog': 'CAROLINA DOG',
  'Petit Basset Griffon Vendeen (PBGV)': 'PBGV',
  'Parson Russell Terrier': 'PARSON RUSS TER',
  'Grand Basset Griffon Vendeen (GBGV)': 'GBGV',
  'St. Bernard (RGH)': 'ST BERNARD RGH',
  'Norfolk Terrier': 'NORFOLK TERRIER',
  'Dachshund (Standard)': 'DACHSHUND',
  'English Shepherd': 'ENG SHEPHERD',
  'Treeing Cur': 'TREEING CUR',
  'Doberman Pinscher': 'DOBERMAN PINSCH',
  'Australian Terrier': 'AUST TERRIER',
  Mastiff: 'MASTIFF',
  'Australian Cattle Dog': 'AUST CATTLE DOG',
  'Golden Retriever': 'GOLDEN RETR',
  'Bernese Mountain Dog': 'BERNESE MTN DOG',
  Schipperke: 'SCHIPPERKE',
  'Finnish Spitz': 'FINNISH SPITZ',
  'Australian Shepherd': 'AUST SHEPHERD',
  'Irish Wolfhound': 'IRISH WOLFHOUND',
  'Border Terrier': 'BORDER TERRIER',
  'Afghan Hound': 'AFGHAN HOUND',
  'Bearded Collie': 'BEARDED COLLIE',
  Newfoundland: 'NEWFOUNDLAND',
  'Collie (Rough)': 'COLLIE ROUGH',
  'Treeing Tennessee Brindle': 'TENN TR BRINDLE',
  'Welsh Corgi (Pembroke)': 'WELSH CORGI PEM',
  Weimaraner: 'WEIMARANER',
  'German Pinscher': 'GERMAN PINSCHER',
  'Old English Sheepdog': 'OLDENG SHEEPDOG',
  Munsterlander: 'MUNSTERLANDER',
  'Dogo Argentino': 'DOGO ARGENTINO',
  Pekingese: 'PEKINGESE',
  'Yorkshire Terrier': 'YORKSHIRE TERR',
  'Australian Kelpie': 'AUST KELPIE',
  'Chihuahua (Short-haired)': 'CHIHUAHUA SH',
  'Curlycoat Retriever': 'CURLYCOAT RETR',
  Whippet: 'WHIPPET',
  'Norwegian Buhund': 'NORW BUHUND',
  'Clumber Spaniel': 'CLUMBER SPAN',
  'Presa Canario': 'PRESA CANARIO',
  Entlebucher: 'ENTLEBUCHER',
  'Neapolitan Mastiff': 'NEAPOLITAN MAST',
  Dalmatian: 'DALMATIAN',
  'Chinese Shar Pei': 'CHINESE SHARPEI',
  'Cane Corso': 'CANE CORSO',
  'Great Pyrenees': 'GREAT PYRENEES',
  'Chow Chow': 'CHOW CHOW',
  Keeshond: 'KEESHOND',
  'Boston Terrier': 'BOSTON TERRIER',
  'American Water Spaniel': 'AMER WATER SPAN',
  'Dandie Dinmont Terrier': 'DANDIE DINMONT',
  Catahoula: 'CATAHOULA',
  'Canaan Dog': 'CANAAN DOG',
  'Tibetan Terrier': 'TIBETAN TERR',
  Eurasier: 'EURASIER',
  'Portuguese Podengo (Pequeno)': 'PODENGO PEQUENO',
  Puli: 'PULI',
  'Anatolian Shepherd': 'ANATOL SHEPHERD',
};
