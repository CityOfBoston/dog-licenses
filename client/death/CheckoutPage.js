// @flow

import React from 'react';
import Head from 'next/head';
import { observer } from 'mobx-react';

import type Cart from '../store/Cart';
import Nav from '../common/Nav';
import CheckoutItem from './checkout/CheckoutItem';

export type InitialProps = {|
|}

export type Props = {
  /* :: ...InitialProps, */
  cart: Cart,
}

@observer
export default class CheckoutPage extends React.Component {
  props: Props;

  handleSubmit = (ev: SyntheticInputEvent) => {
    ev.preventDefault();
  }

  render() {
    const { cart } = this.props;

    return (
      <div>
        <Head>
          <title>Boston.gov — Death Certificate Checkout</title>
        </Head>

        <Nav cart={cart} link="lookup" />

        <div className="p-a300 b--g">
          <div className="sh sh--b0 m-v300">
            <h1 className="sh-title">Checkout</h1>
          </div>
        </div>

        <form className="" acceptCharset="UTF-8" method="get" action="/death" onSubmit={this.handleSubmit}>
          { cart.items.map((item) => <CheckoutItem key={item.id} item={item} cart={cart} />) }

          <div className="p-a300 b--g">
            <button className="btn" type="submit">Pay and Finish</button>
          </div>
        </form>
      </div>
    );
  }
}
