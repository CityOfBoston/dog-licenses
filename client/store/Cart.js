// @flow

import { observable, computed, action } from 'mobx';

import type { DeathCertificate } from '../types';

export class CartItem {
  @observable.ref cert: DeathCertificate;
  @observable quantity: number;
}

export default class Cart {
  @observable items = [];

  @computed get size(): number {
    return this.items.reduce((acc, item) => acc + item.quantity, 0);
  }

  @action add(cert: DeathCertificate, quantity: number) {
    const existingItem = this.items.find((item) => item.cert.id === cert.id);
    if (existingItem) {
      existingItem.quantity += quantity;
    } else {
      const item = new CartItem();
      item.cert = cert;
      item.quantity = quantity;

      this.items.push(item);
    }
  }

  @action setQuantity(certId: string, quantity: number) {
    const existingItem = this.items.find((item) => item.cert.id === certId);
    if (existingItem) {
      existingItem.quantity = quantity;
    }
  }

  @action remove(certId: string) {
    const idx = this.items.findIndex(({ cert }) => cert.id === certId);
    if (idx !== -1) {
      this.items.splice(idx, 1);
    }
  }
}
