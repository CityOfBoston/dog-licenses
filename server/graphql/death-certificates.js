// @flow

import type { Context } from './index';
import type { SearchResult } from '../services/Registry';

export const Schema = `
type DeathCertificate {
  id: String!,
  firstName: String!,
  lastName: String!,
  birthYear: String,
  deathYear: String!,
  causeOfDeath: String,
  age: String,
}

type DeathCertificates {
  search(query: String!): [DeathCertificate!]!
  certificate(id: String!): DeathCertificate
  certificates(ids: [String!]!): [DeathCertificate]!
}
`;

type SearchArgs = {
  query: string,
};

type CertificateArgs = {
  id: string,
};

type CertificatesArgs = {
  ids: string[],
};

type DeathCertificate = {
  id: string,
  firstName: string,
  lastName: string,
  birthYear: ?string,
  deathYear: string,
  causeOfDeath: ?string,
  age: ?string,
};

const TEST_DEATH_CERTIFICATES: DeathCertificate[] = [
  {
    id: '000001',
    firstName: 'Logan',
    lastName: 'Howlett',
    birthYear: '1974',
    deathYear: '2014',
    causeOfDeath: 'Adamantium suffocation',
    age: '058',
  },
  {
    id: '000002',
    firstName: 'Bruce',
    lastName: 'Banner',
    birthYear: '1962',
    deathYear: '2016',
    causeOfDeath: 'Hawkeye',
    age: '61',
  },
  {
    id: '000003',
    firstName: 'Monkey',
    lastName: 'Joe',
    birthYear: '1991',
    deathYear: '2005',
    causeOfDeath: null,
    age: '4 yrs. 2 mos. 10 dys',
  },
];

export const resolvers = {
  DeathCertificates: {
    // eslint-disable-next-line no-unused-vars
    search: async (root: mixed, { query }: SearchArgs, { registry }: Context): Promise<Array<DeathCertificate>> => {
      const results: Array<SearchResult> = await registry.search(query);

      return results.map((res) => ({
        id: res.CertificateID.toString(),
        firstName: res['First Name'],
        lastName: res['Last Name'],
        birthYear: null,
        deathYear: res['Date of Death'].split('/')[2],
        causeOfDeath: null,
        age: res.AgeOrDateOfBirth,
      }));
    },
    certificate: (root: mixed, { id }: CertificateArgs) => TEST_DEATH_CERTIFICATES.find((c) => c.id === id),
    certificates: (root: mixed, { ids }: CertificatesArgs) => ids.map((id) => TEST_DEATH_CERTIFICATES.find((c) => c.id === id)),
  },
};
