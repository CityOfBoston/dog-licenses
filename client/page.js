// @flow

import React from 'react';
import type { Context } from 'next';
import { useStrict } from 'mobx';
import { rehydrate } from 'glamor';
import Router from 'next/router';

import Cart from './store/Cart';
import makeLoopbackGraphql from './loopback-graphql';
import type { LoopbackGraphql } from './loopback-graphql';

import RouterListener from './RouterListener';

// Higher-order component for a Page in our app.

// Pass this mixin a function that requires the component class and returns it.
// require is necessary rather than import since imports are hoisted to the
// beginning of files, and we need to rehydrate before any Glamor "css"
// statements are processed.

export type InitialPropsDependencies = {
  cart: Cart,
  loopbackGraphql: LoopbackGraphql,
}

let browserInited = false;
let cart: Cart;

function maybeInitBrowser() {
  if (browserInited || !process.browser) {
    return;
  }

  browserInited = true;

  // eslint-disable-next-line no-underscore-dangle
  rehydrate(window.__NEXT_DATA__.glamorIds);
  useStrict(true);

  cart = new Cart();
  const routerListener = new RouterListener();
  routerListener.attach(Router);
}

export default <OP, P: $Subtype<Object>, S> (componentFn: () => Class<React.Component<OP, P, S>>): Class<React.Component<void, P, void>> => {
  // Needs to be called before componentFn so we can initialize Glamor before the
  // static "css" constants are evaluated.
  maybeInitBrowser();

  const Component = componentFn();

  return class Page extends React.Component {
    props: P;
    loopbackGraphql: LoopbackGraphql = makeLoopbackGraphql();

    static getInitialProps(ctx: Context<*>) {
      const { req } = ctx;

      const dependencies: InitialPropsDependencies = {
        cart: cart || new Cart(),
        loopbackGraphql: makeLoopbackGraphql(req),
      };

      if (typeof Component.getInitialProps === 'function') {
        return Component.getInitialProps(ctx, dependencies);
      } else {
        return {};
      }
    }

    render() {
      const dependencies = {
        cart: cart || new Cart(),
        loopbackGraphql: this.loopbackGraphql,
      };

      return <Component {...dependencies} {...this.props} />;
    }
  };
};
