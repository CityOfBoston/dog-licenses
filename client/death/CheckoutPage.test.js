// @flow
import React from 'react';
import { runInAction } from 'mobx';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Cart from '../store/Cart';
import { TYPICAL_CERTIFICATE, PENDING_CERTIFICATE } from '../../fixtures/client/death-certificates';

import CheckoutPage from './CheckoutPage';

describe('rendering', () => {
  let cart;

  beforeEach(() => {
    cart = new Cart();
  });

  it('renders loading for unhydrated cart', () => {
    runInAction(() => {
      cart.pendingFetches = 2;
      cart.items = ([{
        id: TYPICAL_CERTIFICATE.id,
        cert: null,
        quantity: 5,
      }, {
        id: PENDING_CERTIFICATE.id,
        cert: null,
        quantity: 3,
      }]: any);
    });

    expect(renderer.create(<CheckoutPage cart={cart} />).toJSON()).toMatchSnapshot();
  });

  it('renders a hydrated cart', () => {
    cart.add(TYPICAL_CERTIFICATE, 5);
    cart.add(PENDING_CERTIFICATE, 2);

    expect(renderer.create(<CheckoutPage cart={cart} />).toJSON()).toMatchSnapshot();
  });
});

describe('submit', () => {
  let cart;
  let wrapper;

  beforeEach(() => {
    cart = new Cart();
    cart.add(TYPICAL_CERTIFICATE, 5);
    cart.add(PENDING_CERTIFICATE, 2);

    wrapper = shallow(<CheckoutPage cart={cart} />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('submits', () => {
    const form = wrapper.find('form');
    const preventDefault = jest.fn();

    form.simulate('submit', { preventDefault });
    expect(preventDefault).toHaveBeenCalled();
  });
});
