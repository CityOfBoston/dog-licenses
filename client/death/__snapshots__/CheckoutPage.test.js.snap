// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders a hydrated cart 1`] = `
<div>
  <nav
    className="nv-s nv-s--sticky"
    data-jsx={3477795240}
  >
    <div
      className="nv-s-l bar"
      data-jsx={3477795240}
    >
      <a
        className="nv-s-l-b back-link"
        data-jsx={3477795240}
        href="/death"
        onClick={[Function]}
      >
        Back to Lookup
      </a>
      <a
        className="cart-link"
        data-jsx={3477795240}
        href="/death/checkout"
        onClick={[Function]}
      >
        7
      </a>
    </div>
  </nav>
  <div
    className="p-a300"
  >
    <div
      className="sh sh--b0 m-v300"
    >
      <h1
        className="sh-title"
      >
        Checkout
      </h1>
    </div>
  </div>
  <form
    acceptCharset="UTF-8"
    action="/death"
    className=""
    method="get"
    onSubmit={[Function]}
  >
    <div
      className="p-a300 result"
      data-jsx={1389666004}
    >
      <input
        aria-label="Quantity"
        className="br br-a150 quantity-box"
        data-jsx={1389666004}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        value="5"
      />
      <div
        className="certificate-info"
        data-jsx={1389666004}
      >
        <div
          data-jsx={1389666004}
        >
          Bruce
           
          Banner
        </div>
        <div
          data-jsx={1389666004}
          style={
            Object {
              "fontStyle": "italic",
            }
          }
        >
          3/4/2016
          <span
            className="id-label"
            data-jsx={1389666004}
          >
            ID:
          </span>
           
          000002
        </div>
      </div>
      <button
        data-jsx={1389666004}
        onClick={[Function]}
        type="button"
      >
        X
      </button>
    </div>
    <div
      className="p-a300 result"
      data-jsx={1389666004}
    >
      <input
        aria-label="Quantity"
        className="br br-a150 quantity-box"
        data-jsx={1389666004}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        value="2"
      />
      <div
        className="certificate-info"
        data-jsx={1389666004}
      >
        <div
          data-jsx={1389666004}
        >
          Logan
           
          Howlett
        </div>
        <div
          data-jsx={1389666004}
          style={
            Object {
              "fontStyle": "italic",
            }
          }
        >
          4/15/2014
          <span
            className="id-label"
            data-jsx={1389666004}
          >
            ID:
          </span>
           
          000001
        </div>
      </div>
      <button
        data-jsx={1389666004}
        onClick={[Function]}
        type="button"
      >
        X
      </button>
    </div>
    <div
      className="p-a300 g"
      data-jsx={1434283171}
    >
      <div
        className="m-v500 g--9"
      >
        <div
          className="t--info"
        >
          7
           
          certificates
           × $
          14
           + 
          2.75
          % credit card fee
        </div>
        <div
          className="sh sh--b0"
        >
          <span
            className="sh-title"
          >
            Subtotal: $
            100.70
          </span>
        </div>
      </div>
      <button
        className="btn g--3 m-v500"
        data-jsx={1434283171}
        type="submit"
      >
        Pay and Finish
      </button>
    </div>
  </form>
</div>
`;

exports[`rendering renders loading for unhydrated cart 1`] = `
<div>
  <nav
    className="nv-s nv-s--sticky"
    data-jsx={3477795240}
  >
    <div
      className="nv-s-l bar"
      data-jsx={3477795240}
    >
      <a
        className="nv-s-l-b back-link"
        data-jsx={3477795240}
        href="/death"
        onClick={[Function]}
      >
        Back to Lookup
      </a>
      <a
        className="cart-link"
        data-jsx={3477795240}
        href="/death/checkout"
        onClick={[Function]}
      >
        8
      </a>
    </div>
  </nav>
  <div
    className="p-a300"
  >
    <div
      className="sh sh--b0 m-v300"
    >
      <h1
        className="sh-title"
      >
        Checkout
      </h1>
    </div>
  </div>
  <form
    acceptCharset="UTF-8"
    action="/death"
    className=""
    method="get"
    onSubmit={[Function]}
  >
    <div
      className="p-a300 g"
      data-jsx={1434283171}
    >
      <div
        className="m-v500 g--9"
      >
        <div
          className="t--info"
        >
          8
           
          certificates
           × $
          14
           + 
          2.75
          % credit card fee
        </div>
        <div
          className="sh sh--b0"
        >
          <span
            className="sh-title"
          >
            Subtotal: $
            115.09
          </span>
        </div>
      </div>
      <button
        className="btn g--3 m-v500"
        data-jsx={1434283171}
        type="submit"
      >
        Pay and Finish
      </button>
    </div>
  </form>
</div>
`;
